### PR TITLE
book entity added, service and repository implemented

### PR DESCRIPTION
**Pull request by TetianaDenchyk**

### CODE REVIEW

**Review by pastcwdt — APPROVED** [review]
Good solution!

File: src/main/java/com/bookshop/exception/BookNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.bookshop.exception;
+
+public class BookNotFoundException extends RuntimeException {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/exception/BookNotFoundException.java]: Don’t customize your Exceptions in this case you need to create exceptions for each entity.
```suggestion
public class EntityNotFoundException extends RuntimeException {

```

File: src/main/java/com/bookshop/exception/BookNotSavedException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.bookshop.exception;
+
+public class BookNotSavedException extends RuntimeException {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/exception/BookNotSavedException.java]:
```suggestion
public class DataProcessingException extends RuntimeException {

```

File: src/main/java/com/bookshop/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/model/Book.java]: Lombok is a great library but be [attentive when using it with Entity](https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/) JFYI

File: src/main/java/com/bookshop/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/model/Book.java:33]:
```suggestion
``` 
Redundant  empty line

File: src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java]: As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean defines only one constructor to begin with. See [note here ](https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html)Let’s check it ;)

File: src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            return book;
+        } catch (Exception e) {
+            throw new BookNotSavedException("Can't save book to the DB, book: " + book, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book b", Book.class).getResultList();
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java]: you don’t  need to use an alias here

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good! Let’s  fix a few mistakes ;)

File: src/main/java/com/bookshop/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    private String author;
+
+    @Column(nullable = false, unique = true)
+    private String isbn;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    private String description;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/model/Book.java:33]:
```suggestion
```

File: src/main/java/com/bookshop/service/impl/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import com.bookshop.service.BookService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/BookServiceImpl.java:14]:
```suggestion
```

**Review by liliia-ponomarenko — APPROVED** [review]

**Pull request merged**